### PR TITLE
Docs: Add code snippets for yarn pnp in migration-guide

### LINKED ISSUE #21895: [Bug]: `storybook init` (v7) fails with Yarn PnP
### Describe the bug

Running `storybook init` with Yarn PnP causes this error:

```
> npx storybook init

 storybook init - the simplest way to add a Storybook to your project. 

(node:38504) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
 • Detecting project type. ✓
    Detected webpack project. Setting builder to webpack
 • Adding Storybook support to your "Create React App" based project➤ YN0000: ┌ Resolution step
➤ YN0002: │ @storybook/preset-create-react-app@npm:7.0.2 [1b0ac] doesn't provide react-refresh (paa6d3), requested by @pmmmwh/react-refresh-webpack-plugin
➤ YN0002: │ @storybook/preset-create-react-app@npm:7.0.2 [1b0ac] doesn't provide typescript (pe909d), requested by @storybook/react-docgen-typescript-plugin
➤ YN0002: │ @storybook/preset-create-react-app@npm:7.0.2 [1b0ac] doesn't provide webpack (p28ecc), requested by @pmmmwh/react-refresh-webpack-plugin
➤ YN0002: │ @storybook/preset-create-react-app@npm:7.0.2 [1b0ac] doesn't provide webpack (pd0ace), requested by @storybook/react-docgen-typescript-plugin
➤ YN0002: │ before-storybook@workspace:. doesn't provide @babel/core (p55c72), requested by @storybook/preset-create-react-app
➤ YN0002: │ before-storybook@workspace:. doesn't provide @testing-library/dom (p1ac37), requested by @testing-library/user-event
➤ YN0002: │ eslint-config-react-app@npm:7.0.1 [dfa48] doesn't provide @babel/plugin-syntax-flow (p2f44f), requested by eslint-plugin-flowtype
➤ YN0002: │ eslint-config-react-app@npm:7.0.1 [dfa48] doesn't provide @babel/plugin-transform-react-jsx (pd12b2), requested by eslint-plugin-flowtype
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed in 11s 415ms
➤ YN0000: ┌ Fetch step
➤ YN0013: │ write-file-atomic@npm:2.4.3 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ write-file-atomic@npm:4.0.2 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ ws@npm:6.2.2 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ xtend@npm:4.0.2 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yauzl@npm:2.10.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0000: └ Completed in 2s 441ms
➤ YN0000: ┌ Link step
➤ YN0000: │ ESM support for PnP uses the experimental loader API and is therefore experimental
➤ YN0007: │ esbuild@npm:0.17.15 must be built because it never has been before or the last one failed
➤ YN0000: └ Completed in 1s 878ms
➤ YN0000: Done with warnings in 16s 244ms
. ✓
 • Preparing to install dependencies. ✓


 • Installing dependencies➤ YN0000: ┌ Resolution step
➤ YN0002: │ @storybook/preset-create-react-app@npm:7.0.2 [1b0ac] doesn't provide react-refresh (paa6d3), requested by @pmmmwh/react-refresh-webpack-plugin
➤ YN0002: │ @storybook/preset-create-react-app@npm:7.0.2 [1b0ac] doesn't provide typescript (pe909d), requested by @storybook/react-docgen-typescript-plugin
➤ YN0002: │ @storybook/preset-create-react-app@npm:7.0.2 [1b0ac] doesn't provide webpack (p28ecc), requested by @pmmmwh/react-refresh-webpack-plugin
➤ YN0002: │ @storybook/preset-create-react-app@npm:7.0.2 [1b0ac] doesn't provide webpack (pd0ace), requested by @storybook/react-docgen-typescript-plugin
➤ YN0002: │ before-storybook@workspace:. doesn't provide @babel/core (p55c72), requested by @storybook/preset-create-react-app
➤ YN0002: │ before-storybook@workspace:. doesn't provide @testing-library/dom (p1ac37), requested by @testing-library/user-event
➤ YN0002: │ eslint-config-react-app@npm:7.0.1 [dfa48] doesn't provide @babel/plugin-syntax-flow (p2f44f), requested by eslint-plugin-flowtype
➤ YN0002: │ eslint-config-react-app@npm:7.0.1 [dfa48] doesn't provide @babel/plugin-transform-react-jsx (pd12b2), requested by eslint-plugin-flowtype
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed in 0s 421ms
➤ YN0000: ┌ Fetch step
➤ YN0000: └ Completed in 0s 761ms
➤ YN0000: ┌ Link step
➤ YN0000: │ ESM support for PnP uses the experimental loader API and is therefore experimental
➤ YN0000: └ Completed in 0s 390ms
➤ YN0000: Done with warnings in 2s 3ms
. ✓
🔎 checking possible migrations..

attention => Storybook now collects completely anonymous telemetry regarding usage.
This information is used to shape Storybook's roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://storybook.js.org/telemetry

[Storybook automigrate] ❌ Failed trying to evaluate .storybook\main.js with the following error: Cannot find module '@storybook\addon-links\package.json'
Require stack:
- C:\Users\nbier\Documents\test-storybook-init-yarn-pnp\.storybook\main.js
- C:\Users\nbier\AppData\Local\npm-cache\_npx\6fe9a9991b157df1\node_modules\@storybook\core-common\dist\index.js
- C:\Users\nbier\AppData\Local\npm-cache\_npx\6fe9a9991b157df1\node_modules\@storybook\telemetry\dist\index.js
- C:\Users\nbier\AppData\Local\npm-cache\_npx\6fe9a9991b157df1\node_modules\@storybook\cli\dist\generate.js
- C:\Users\nbier\AppData\Local\npm-cache\_npx\6fe9a9991b157df1\node_modules\@storybook\cli\bin\index.js
Please fix the error and try again.
> 
```

### To Reproduce

1. Clone https://github.com/Methuselah96/test-storybook-init-yarn-pnp.
2. Run `npx storybook init`.

### System

```shell
Environment Info:

  System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
  Binaries:
    Node: 18.12.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - C:\Program Files\nodejs\yarn.CMD
    npm: 8.19.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.1413.0), Chromium (111.0.1661.62)
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jackw**
Closes #21895

## What I did

Updates the docs to include commands for yarn pnp for migrating.

## Checklist

- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [x] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: docs/migration-guide.md
Comment (jonniebigodes) [inline docs/migration-guide.md:28]: @jackw can you update not only the snippet's naming per #21918 but also the command you've used as well so that we provide the correct information?

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@jackw thanks for taking the time to create this pull request and help us improve our documentation. We appreciate it. I left a small item that requires your attention and this should be good to go.

Let me know once you've addressed it and we'll take it from there.


Hope you have a great day.

Stay safe

File: docs/migration-guide.md
Comment (jackw) [reply docs/migration-guide.md:28]: Ah ok, I missed that. I'll rebase, fix conflicts and update the commands as requested. Thanks for the feedback @jonniebigodes ! 🙏

Comment (jackw) [reply docs/migration-guide.md:28]: Actually, it looks like this is targetting `next` but the PRs mentioned are targeting `main`. 🤔

Comment (jonniebigodes) [reply docs/migration-guide.md:28]: Indeed you're right; those were targeting the `main` branch. But you don't need to worry about it. Once this is merged, it will be patched up to the `main` branch.

Comment (jackw) [reply docs/migration-guide.md:28]: Ah ok cool. I'll sort it out shortly. Thanks for getting back to me so quickly.

**Comment by jackw** [comment]
>Let me know once you've addressed it and we'll take it from there.

@jonniebigodes I've updated the PR as requested. Let me know if you need anything more from my side.

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@jackw sorry for the delayed response 🙏 . I just have one final item for you to address so that we can merge this pull request. 

Let me know once you have addressed the feedback and I'll gladly merge it.

Hope you have a great week

Stay safe

File: docs/snippets/common/init-command.yarn.js.mdx
Comment (jackw) [reply docs/snippets/common/init-command.yarn.js.mdx]: @jonniebigodes Thanks for getting back to me on this. I've updated this init command and the upgrade command to use `latest` instead of `next`. 👍

Comment (jonniebigodes) [reply docs/snippets/common/init-command.yarn.js.mdx]: No need to thank whatsoever. My pleasure, one final thing, if you could also update the `storybook-upgrade-prerelease.yarn.js.mdx` to use the same structure (i.e., `latest´ instead of `next`), I'd appreciate it. As for now, we're recommending it to avoid any issues.

**Review by jonniebigodes — APPROVED** [review]

**Comment by jonniebigodes** [comment]
@jackw Thanks for making the changes so promptly. I just got the notification about the latest commit. I'm going to merge this pull request once the checklist clears. Once again, thank you so much for taking the time to put together this pull request and help us improve our documentation 🙏

**Pull request merged**
